server/src/main/java/io/druid/client/ImmutableDruidServer.java

```diff
@@ -128,4 +128,29 @@ public String toString()
            + "', sources='" + dataSources
            + "'}";
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    DruidServer that = (DruidServer) o;
+
+    if (getName() != null ? !getName().equals(that.getName()) : that.getName() != null) {
```

Comment:
Please annotate `DruidServerMetadata.getName()` as `@Nullable`.

Please use `Objects.equals()`, or regenerate equals and hashCode using intelliJ

Reply:
I think it's not nullable because ```DruidServer``` doesn't check in ```compareTo()``` and ```ImmutableDruidServer``` is created from ```DruidServer```. Added null check in the constructor of ```DruidServerMetadata```.

Also changed to use ```DruidServerMetadata``` in ```equals()``` and ```hashCode()``` as in https://github.com/druid-io/druid/pull/5056.